app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,23 @@
+class MessagesController < ApplicationController
+
+  def create
+    @message = Message.new(message_params)
+    @message.user = current_user
+    @message.conversation = Conversation.find(params[:conversation_id])
+    if @message.save
+      redirect_to conversation_path(@message.conversation, anchor: @message.id)
+    else
+      redirect_to conversation_path(@message.conversation, anchor: 'new_message', flash: {error: @message.errors.full_messages.join(', ')})
```

Comment:
You're putting errors in your flash hash, but you're not displaying them on the view anywhere. It's very similar to how we processed and presented errors in P2. [Read this for more.](http://guides.rubyonrails.org/action_controller_overview.html#the-flash)